activerecord/lib/active_record/relation/predicate_builder.rb

```diff
@@ -40,9 +42,12 @@ def expand(column, value)
       # For polymorphic relationships, find the foreign key and type:
       # PriceEstimate.where(estimate_of: treasure)
       if table.associated_with?(column)
-        value = AssociationQueryValue.new(table.associated_table(column), value)
+        if polymorphic_array?(value)
```

Comment:
This logic feels kinda strange here to me. Can we move this conditional into a factory method on `AssociationQueryHandler`?